model_io/urdf/src/urdf_generic_sensor_import.cpp

```diff
@@ -0,0 +1,282 @@
+        
+        std::string linkOption, jointOption;
+        
+        if(!parent)
+        { 
+            std::cerr<<"parent object not found \n";
+            returnVal = false;
+            break;
+        }
+        if(parent->Attribute("link")!= NULL)
+        {
+            linkOption = std::string(parent->Attribute("link"));
+        }
+        else if(parent->Attribute("joint")!=NULL)
+        {
+            jointOption = std::string(parent->Attribute("joint"));
+        }
+        else
+        {
+                std::cerr<<"link or joint attribute for parent object not found \n";
+                returnVal = false;
+                break;
+        } 
+        
+        if(!linkOption.empty())
+        {
+              newGenericSensor.parentObject = iDynTree::GenericSensorData::LINK;
+              newGenericSensor.parentObjectName = linkOption;
+        }
+        else
+        {
+              newGenericSensor.parentObject = iDynTree::GenericSensorData::JOINT;
+              newGenericSensor.parentObjectName = jointOption;
+        }                
+         
+        TiXmlElement* originTag = sensorXml->FirstChildElement("origin");
+        if( originTag )
+        {             
+#ifdef DEBUG
+            std::cout<<"originTag exists \n";
+            std::string  poseText = originTag->GetText();
+            std::cout<<"originTag : \n"<<poseTag->ToText();
+            std::cout<<"Pose rpy:"<<rpyText.c_str()<<", xyz"<<xyzText.c_str()<<"\n";
+#endif //DEBUG
+            std::string  rpyText(originTag->Attribute("rpy"));
+            std::string  xyzText(originTag->Attribute("xyz"));
+            
+            std::vector<std::string> rpyElems;
+            std::vector<std::string> xyzElems;
+            split(rpyText,rpyElems);
+            split(xyzText,xyzElems);
+#ifdef DEBUG            
+            std::cout<<"Pose rpy:"<<rpyElems[0].c_str()<<","<<rpyElems[1].c_str()<<","<<rpyElems[2].c_str()<<", xyz";
+            std::cout<<xyzElems[0].c_str()<<","<<xyzElems[1].c_str()<<","<<xyzElems[2].c_str()<<","<<"\n";
+#endif //DEBUG
+            if( rpyElems.size() != 3 && xyzElems.size() !=3 )
+            {
+                std::cerr<<"Pose attribute for sensor specified incorrectly";
+               returnVal = false;
+            }
+            double roll  = atof(rpyElems[0].c_str());
+            double pitch = atof(rpyElems[1].c_str());
+            double yaw   = atof(rpyElems[2].c_str());
+            iDynTree::Rotation rot = iDynTree::Rotation::RPY(roll,pitch,yaw);
+            double x  = atof(xyzElems[0].c_str());
+            double y = atof(xyzElems[1].c_str());
+            double z   = atof(xyzElems[2].c_str());
+            iDynTree::Position pos(x,y,z);
+            newGenericSensor.sensorPose = iDynTree::Transform(rot,pos); 
+        }
+        else
+        {
+            // default value
+            newGenericSensor.sensorPose = iDynTree::Transform();
```

Comment:
There is also a function `iDynTree::Transform::Identity()` to express more clearly the the default transform is the identity.